csrc/tma.cpp

```diff
@@ -0,0 +1,391 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <tma.h>
+
+#include <cuda_utils.h>
+#include <expr_evaluator.h>
+#include <fusion.h>
+#include <ir/all_nodes.h>
+#include <ir/builder.h>
+#include <utils.h>
+
+#include <cstdint>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+namespace nvfuser {
+
+namespace tma {
+
+#if (CUDA_VERSION >= 12000)
```

Comment:
Perhaps this conditions should be changed to `11800` (if I'm not wrong), as Hopper support has been introduced starting from CTK 11.8 ([link](https://docs.nvidia.com/cuda/hopper-compatibility-guide/index.html#applications-built-using-cuda-toolkit-11-8)).

Reply:
I can not find `cuTensorMapEncodeTiled` in CUDA 11.8's documentation: https://docs.nvidia.com/cuda/archive/11.8.0/

Reply:
Ok, so let's keep it safe and stay with CTK 12.0.

Thanks!